src/Http/Http.Abstractions/src/HttpProtocol.cs

```diff
@@ -102,6 +118,7 @@ public static string GetHttpProtocol(Version version)
                 { Major: 2, Minor: 0 } => Http2,
                 { Major: 1, Minor: 1 } => Http11,
                 { Major: 1, Minor: 0 } => Http10,
+                { Major: 0, Minor: 9 } => Http09,
                 _ => throw new ArgumentOutOfRangeException(nameof(version), "Version doesn't map to a known HTTP protocol.")
```

Comment:
Just as thought: don't add `Http09` here, but throw a `ProtocolNotSupportedException`, so that this exception can be handled upstreams with a concise message why the request is failing.

With the current state it looks like HTTP/0.9 is supported, which it actually isn't at all.

Reply:
>looks like HTTP/0.9 is supported

Good point, we have several cases here:
1. `IIS` inproc hosting actually supports `HTTP\0.9`. If `HostFilteringMiddleware` is not used, you will get content.
2.  If `HostFilteringMiddleware` is used, it returns host validation failed HTML response (without status code in the first line as in `HTTP/1.0` or higher)
2. `Kestrel` doesn't support it and returns `HTTP/1.1 400 Bad Request`

I agree field `Http09` can be misleading. Maybe put extra notes in the comment?

I don't think we can throw. It is a regression and before regression `HTTP\0.9` was parsed successfully.

Reply:
Thanks for your points!

> Maybe put extra notes in the comment?

That would make it more obvious (for the future).